Sources/MapboxNavigation/NavigationMapView+BuildingHighlighting.swift

```diff
@@ -15,13 +15,122 @@ extension NavigationMapView {
      - returns: Bool indicating if number of buildings found equals number of coordinates supplied.
      */
     @discardableResult public func highlightBuildings(at coordinates: [CLLocationCoordinate2D], in3D extrudesBuildings: Bool = true) -> Bool {
-        return false
+        let _ = coordinates.map({buildingIdentifier(at: $0, in3D: extrudesBuildings) })
+        DispatchQueue.main.asyncAfter(deadline: .now() + handleAsyncTime, execute: {self.addBuildingsLayer(with: self.foundBuildingIds, in3D: extrudesBuildings)})
+        return foundBuildingIds.count == coordinates.count
     }
     
     /**
      Removes the highlight from all buildings highlighted by `highlightBuildings(at:in3D:)`.
      */
     public func unhighlightBuildings() {
-        
+        guard let _ = try? mapView.style.getLayer(with: IdentifierString.buildingExtrusionLayer, type: FillExtrusionLayer.self).get() else { return }
+        foundBuildingIds = Set<Double>()
+        let _ = mapView.style.removeStyleLayer(forLayerId: IdentifierString.buildingExtrusionLayer)
     }
+
+    private func featureIdentifierToDouble(features: [Feature]) -> Double? {
+        if let featureIdentifier = features.first?.identifier {
+            switch featureIdentifier {
+            case .string(let stringId): return Double(stringId)
+            case .number(let numberId):
+                switch numberId {
+                case .double(let doubleId): return Double(doubleId)
+                case .int(let intId): return Double(intId)
+                }
+            }
+        }
+        return nil
+    }
+
+    private func buildingIdentifier(at coordinate: CLLocationCoordinate2D, in3D extrudesBuildings: Bool = true) {
```

Comment:
Do we need to use `extrudesBuildings` here?

Reply:
Thanks for the suggestion! I'll remove this part.